backend/webserver/views.py

```diff
@@ -268,6 +271,34 @@ def get(self, request):
         return Response(serializer.data, status=status.HTTP_200_OK)
 
 
+class AuthorLikedView(APIView):
+    authentication_classes = [BasicAuthentication]
+    
+    def get(self,requst,pk):
+        author = get_object_or_404(Author,pk=pk)
+        likes = Like.objects.filter(post__visibility="PUBLIC").filter(author=author)
+        serializer = PostLikeSerializer(likes, many=True, context={'request': requst})
+        # I am doing this so that I can display the post like this "object":"http://127.0.0.1:5454/authors/9de17f29c12e8f97bcbbd34cc908f1baba40658e/posts/764efa883dda1e11db47671c4a3bbd9e"
+        for data in serializer.data: 
+            data['post'] = urljoin(data['author']['url'],  f"posts/{data['post']}/")
+            
+        return Response(serializer.data, status=status.HTTP_200_OK)
+        
+
+class PostLikesView(APIView):
+    authentication_classes = [BasicAuthentication]
+    
+    def get(self,requst,pk,post_id):
+        author = get_object_or_404(Author,pk=pk)
+        post = get_object_or_404(Post,id=post_id,author=author.id)
+        likes = Like.objects.filter(post=post.id)
+        serializer = PostLikeSerializer(likes, many=True, context={'request': requst})
+        # I am doing this so that I can display the post like this "object":"http://127.0.0.1:5454/authors/9de17f29c12e8f97bcbbd34cc908f1baba40658e/posts/764efa883dda1e11db47671c4a3bbd9e"
+        for data in serializer.data: 
+            data['post'] = urljoin(data['author']['url'],  f"posts/{data['post']}/")
```

Comment:
I made a similar comment below, but this work should happen at the `PostLikeSerializer`. this is fine for now, but i would refactor it later